python/ray/data/_internal/execution/interfaces/op_runtime_metrics.py

```diff
@@ -568,7 +662,30 @@ def on_task_finished(self, task_index: int, exception: Optional[Exception]):
                     assert meta.size_bytes is not None
                     self.obj_store_mem_spilled += meta.size_bytes
 
+                    if self._per_node_metrics_enabled:
+                        if meta.exec_stats.node_id is not None:
+                            node_metrics = self._per_node_metrics[
+                                meta.exec_stats.node_id
+                            ]
+                        else:
+                            node_metrics = self._per_node_metrics[_NODE_UNKNOWN]
```

Comment:
That's actually a great idea to assure consistency of the metrics even in cases when metadata won't be available!